user-guide/Advanced_Functionality/Databases/STaaS/STaaS.md

```diff
@@ -21,13 +16,96 @@ Advantages of DataMiner Storage as a Service (STaaS) include:
+
+1. Make sure you have at least **DataMiner CloudGateway 2.8.0** installed on the system. See [Upgrading nodes to the latest DxM versions](xref:Managing_cloud-connected_nodes#upgrading-nodes-to-the-latest-dxm-versions).
+
+1. Contact your Skyline representative or <staas@dataminer.services> to register you system to use STaaS.
+
+1. Wait until the registration is completed.
+
+1. **Optionally**, contact your Skyline representative or <staas@dataminer.services> to migrate your existing data to STaaS.
+
+1. In the `C:\Skyline DataMiner` folder, open *DB.xml* and edit it to look like this:
+
+   ```xml
+   <?xml version="1.0"?>
+   <DataBases xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns="http://www.skyline.be/config/db">
+      <DataBase active="true" local="true" search="true" cloud="true" type="CloudStorage"/>
+   </DataBases>
+   ```
+
+1. Restart your DMA and STaaS will be used.
+
+## Limitations
+
+To **migrate existing data** to STaaS, the following limitations apply:
+
+- If you migrate from a Cassandra database, logger tables cannot be migrated yet. Support for the migration of logger tables will be introduced in a future DataMiner version.
+- Migration of an Elasticsearch database that is defined in [DBConfiguration.xml](xref:DBConfiguration_xml) instead of [DB.xml](xref:DB_xml) (e.g. because a setup with multiple Elasticsearch clusters is used) is not yet supported.
+- Migration from a MySQL setup is not yet supported.
+
+In addition, the following **other limitations** currently apply:
+
+- [Jobs](xref:jobs), [Ticketing](xref:ticketing), and [API Deployment](xref:Overview_of_Soft_Launch_Options#apideployment) data are not supported.
+- Indexing engine functionality such as search, suggestions, aliases, and aggregation are not supported.
+- Direct queries from DataMiner Cube to the database are not supported.
+- The [SLReset tool](xref:Factory_reset_tool) is not supported.
+- [Exporting trend data](xref:Exporting_elements_services_etc_to_a_dmimport_file) to a .dmimport file is not supported.
+- Proxy and DMZ setups are currently not supported.
+- The [autoincrement](xref:Protocol.Params.Param.ArrayOptions.ColumnOption-type#autoincrement) tag on logger tables is not supported.
+- DOM queries can be slower depending on the number of DOM records and the complexity of the query. This limitation will be removed in the near future.
+
+## Troubleshooting
+
+> [!NOTE]
+> If you experience any issues during setup or while using Storage as a Service, and you cannot resolve these with the procedures below, contact <staas@dataminer.services>.
+
+### Cloud connection lost
+
+Under normal circumstances, CloudGateway refreshes the cloud session automatically. However, if CloudGateway is down for longer than three days, for example because the server is down, the cloud session will become invalid. This will cause DataMiner startup to fail.
```

Comment:
What is the most common symptom of this issue in the DataMiner logging? As users may search for that when troubleshooting, it would help if we add this.

Reply:
the user can see an exception in both SLDBConnection.txt or SLCloudStorage.txt saying that the token is expired

Reply:
Do you know what this exception looks like exactly? I'd like to add this, because users will most likely search for that specific text when they encounter the issue.

Reply:
this is the exception:

```
2023/10/02 14:04:23.458|SLDBConnection|SLCloudStorage|INF|0|6|2023-10-02T14:04:23.442|FATAL|DataGateway.CloudStorage.CloudStorage|CloudSettings could not be retrieved from the cloud. Retrying in 00:00:05. Exception: SLCloudStorage.Repositories.Exceptions.CloudSettingsRepositoryException: Failed to do GetCloudAccessTokenRequest. Received the following error messages: { "message": "The Service Principal of this DMS is expired (3/14/2023 8:09:51 AM +00:00) but should soon be refreshed automatically." }
   at SLCloudStorage.Repositories.Impl.HttpCloudSettingsRepository.ExecuteBroker[T](Func`1 action, Func`2 getError, String message) in C:\devcore3\server\SLDataGateway\SLCloudStorage\Repositories\Impl\HttpCloudSettingsRepository.cs:line 230
   at SLCloudStorage.Repositories.Impl.HttpCloudSettingsRepository.GetToken(String scope) in C:\devcore3\server\SLDataGateway\SLCloudStorage\Repositories\Impl\HttpCloudSettingsRepository.cs:line 144
   at SLCloudStorage.Repositories.Impl.HttpCloudSettingsRepository.<GetAsync>d__4.MoveNext() in C:\devcore3\server\SLDataGateway\SLCloudStorage\Repositories\Impl\HttpCloudSettingsRepository.cs:line 57
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at SLCloudStorage.CloudStorageManager.<LoadSettingsAsync>d__62.MoveNext() in C:\devcore3\server\SLDataGateway\SLCloudStorage\CloudStorageManager.cs:line 355)
```

Reply:
also, this one can also occour when CloudGateway is not running:

`2023/10/02 14:05:33.981|SLDBConnection|SLCloudStorage|INF|0|6|2023-10-02T14:05:33.980|FATAL|DataGateway.CloudStorage.CloudStorage|CloudSettings could not be retrieved from the cloud. Retrying in 00:00:05. Exception: SLCloudStorage.Repositories.Exceptions.CloudSettingsRepositoryException: Exception while doing GetCcaGatewayConfigRequest. ---> System.AggregateException: One or more errors occurred. ---> DataMinerMessageBroker.API.Exceptions.SubscriptionException: No responders are available for the request. ---> NATS.Client.NATSNoRespondersException: No responders are available for the request.
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NATS.Client.Connection.<RequestAsyncImpl>d__147.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at DataMinerMessageBroker.API.Nats.NatsSession.<RequestAsync>d__50.MoveNext()
   --- End of inner exception stack trace ---
   at DataMinerMessageBroker.API.Nats.NatsSession.<RequestAsync>d__50.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at DataMinerMessageBroker.API.MessageBroker.<RequestAsync>d__17.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at SLCloudStorage.Repositories.Impl.HttpCloudSettingsRepository.<GetApiEndpoint>b__6_0() in C:\devcore3\server\SLDataGateway\SLCloudStorage\Repositories\Impl\HttpCloudSettingsRepository.cs:line 117
   at SLCloudStorage.Repositories.Impl.HttpCloudSettingsRepository.ExecuteBroker[T](Func`1 action, Func`2 getError, String message) in C:\devcore3\server\SLDataGateway\SLCloudStorage\Repositories\Impl\HttpCloudSettingsRepository.cs:line 216
   --- End of inner exception stack trace ---
   at SLCloudStorage.Repositories.Impl.HttpCloudSettingsRepository.ExecuteBroker[T](Func`1 action, Func`2 getError, String message) in C:\devcore3\server\SLDataGateway\SLCloudStorage\Repositories\Impl\HttpCloudSettingsRepository.cs:line 220
   at SLCloudStorage.Repositories.Impl.HttpCloudSettingsRepository.GetApiEndpoint() in C:\devcore3\server\SLDataGateway\SLCloudStorage\Repositories\Impl\HttpCloudSettingsRepository.cs:line 113
   at SLCloudStorage.Repositories.Impl.HttpCloudSettingsRepository.<GetAsync>d__4.MoveNext() in C:\devcore3\server\SLDataGateway\SLCloudStorage\Repositories\Impl\HttpCloudSettingsRepository.cs:line 46
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at SLCloudStorage.CloudStorageManager.<LoadSettingsAsync>d__62.MoveNext() in C:\devcore3\server\SLDataGateway\SLCloudStorage\CloudStorageManager.cs:line 355)`